xarray/backends/api.py

```diff
@@ -218,10 +218,12 @@ def check_attr(name, value):
                 "serialization to netCDF files"
             )
 
-        if not isinstance(value, (str, Number, np.ndarray, np.number, list, tuple)):
+        if not isinstance(
+            value, (str, Number, np.ndarray, np.number, np.bool_, list, tuple)
```

Comment:
@caenrigen I'm also not an expert in backends. But shouldn't we allow `np.bool_`  here only in the case of `h5netcdf`-backend with `invalid_netcdf=True`? So that for users who are about to serialize `np.bool_` to `netCDF` the TypeError is raised?

Reply:
Thank you for the review, well spotted, indeed a more careful handling may be needed, will look into this

Reply:
Hi! Got it fixed to limit the check to the specific use-case. Thank you for the review 🙂